modules/core/shared/src/main/scala/gem/TargetEnvironment.scala

```diff
@@ -4,65 +4,188 @@
 package gem
 
 import cats.Eq
-import cats.implicits._
 import gem.enum.Instrument
 import scala.collection.immutable.TreeSet
 
 /** Collection of targets associated with an observation. */
 sealed trait TargetEnvironment {
-
-  type I <: Instrument with Singleton
-
-  val asterism: Option[Asterism.Aux[I]]
-
-  val userTargets: TreeSet[UserTarget]
-
+  def asterism: Option[Asterism]
+  def userTargets: TreeSet[UserTarget]
```

Comment:
Guiding will be instrument-specific as well in order to control which guide probes may be assigned.  It will need some thought to figure out how the PWFSs fit in.

Reply:
Yeah I'm not sure what goes where but at least now everything is instrument-specific so we have some wiggle room.